### PR TITLE
Extend 102 to excepts, 104 also catch yield

### PR DESCRIPTION
**Pull request by jakkdl**
Also add info to error messages on 102&103, rename variables and minor refactor of Visitor102 to match later visitors.

### CODE REVIEW

File: flake8_trio.py [hunk 1]
```diff
@@ -37,7 +37,7 @@ def make_error(error: str, lineno: int, col: int, *args: Any, **kwargs: Any) ->
 
 
 class Flake8TrioVisitor(ast.NodeVisitor):
-    def __init__(self) -> None:
```
Comment (jakkdl) [inline flake8_trio.py:40]: I was originally also running my code with `mypy --strict` - which is much stricter than pyright and also enforces explicit return values. But I transitioned to only running the code through pyright and especially usage of ` -> None` became inconsistent, so I thought it best to just clean up all the old instances of it

File: flake8_trio.py [hunk 2]
```diff
@@ -185,112 +188,153 @@ def visit_Import(self, node: ast.Import):
...
 class Visitor102(Flake8TrioVisitor):
-    def __init__(self) -> None:
+    def __init__(self):
         super().__init__()
-        self._inside_finally: bool = False
-        self._scopes: List[TrioScope] = []
-        self._context_manager = False
+        self._critical_scope: Optional[Tuple[int, int, str]] = None
+        self._trio_context_managers: List[TrioScope] = []
+        self._safe_decorator = False
```
Comment (jakkdl) [inline flake8_trio.py:226]: I thought through these names again, and realized that both decorators and `with` are context managers, so I think these names are better.

File: flake8_trio.py [hunk 3]
```diff
@@ -185,112 +188,153 @@ def visit_Import(self, node: ast.Import):
...
 
-    def visit_Assign(self, node: ast.Assign) -> None:
```
Comment (jakkdl) [inline flake8_trio.py:195]: moved down since it's not the key part of the class

File: flake8_trio.py [hunk 4]
```diff
@@ -596,8 +622,8 @@ def run(self) -> Generator[Tuple[int, int, str, Type[Any]], None, None]:
 
 TRIO100 = "TRIO100: {} context contains no checkpoints, add `await trio.sleep(0)`"
 TRIO101 = "TRIO101: yield inside a nursery or cancel scope is only safe when implementing a context manager - otherwise, it breaks exception handling"
-TRIO102 = "TRIO102: it's unsafe to await inside `finally:` unless you use a shielded cancel scope with a timeout"
-TRIO103 = "TRIO103: except Cancelled or except BaseException block with a code path that doesn't re-raise the error"
+TRIO102 = "TRIO102: await inside {2} @ line,col {0},{1} must have shielded cancel scope with a timeout"
+TRIO103 = "TRIO103: {} block with a code path that doesn't re-raise the error"
 TRIO104 = "TRIO104: Cancelled (and therefore BaseException) must be re-raised"
 TRIO105 = "TRIO105: Trio async function {} must be immediately awaited"
 TRIO106 = "TRIO106: trio must be imported with `import trio` for the linter to work"
```
Comment (jakkdl) [inline flake8_trio.py:629]: Finally got around to polishing up error messages a bit more, though unsure of exact format of `inside BaseException @ line,col 120, 4`. Column is likely overkill?
101 and 104 can get the same treatment, but didn't wanna update all the messages until settled.

Comment (Zac-HD) [reply flake8_trio.py:629]: "inside X on line Y" seems good to me, column information doesn't seem worth the verbosity.

File: tests/test_flake8_trio.py [hunk 5]
```diff
@@ -40,9 +40,19 @@ def trim_messages(messages: Iterable[Error]):
         errors = tuple(plugin.run())
 
         # start with a check with trimmed errors that will make for smaller diff messages
-        trim_errors = trim_messages(plugin.run())
+        trim_errors = trim_messages(errors)
         trim_expected = trim_messages(expected)
-        self.assertTupleEqual(trim_errors, trim_expected)
+
+        #
```
Comment (Zac-HD) [inline tests/test_flake8_trio.py:46]: missing comment

File: tests/trio102.py [hunk 6]
```diff
@@ -122,3 +122,39 @@ async def foo3():
...
+
+
+async def foo5():
+    await foo()  # avoid TRIO107
+    try:
+        ...
+    except trio.Cancelled:
+        with trio.CancelScope(deadline=30, shield=True):
+            await foo()  # safe
+            raise  # avoid TRIO103
```
Comment (Zac-HD) [inline tests/trio102.py:152]: dedent these for clarity?  the `raise` doesn't need to be inside the cancelscope.

**Review by Zac-HD — COMMENTED** [review]
Two small suggestions below and one above; once those are handled it's ready to merge 🎉

File: tests/test_flake8_trio.py (hunk 5 above)
Comment (jakkdl) [reply tests/test_flake8_trio.py:46]: This function is getting a makeover in #19 (which has merged this branch), so I'd leave it for the moment and avoid a merge conflict.

File: tests/trio102.py (hunk 6 above)
Comment (jakkdl) [reply tests/trio102.py:152]: Actually, lemme fix this in #19 too (I'm touching all those lines and getting rid of the `raise`s completely).

Comment (jakkdl) [reply tests/trio102.py:152]: nvm no I'm not. 1 sec I'll fix

**Pull request merged**